src/features/session_replay/aggregate/index.js

```diff
@@ -112,23 +102,34 @@ export class Aggregate extends AggregateBase {
       }
     }, this.featureName, this.ee)
 
+    // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
+    // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
+    registerHandler('sr-errorAgg', (e) => {
+      this.errorNoticed = true
+      if (this.recorder) this.recorder.currentBufferTarget.hasError = true
+      // run once
+      if (this.mode === MODE.ERROR && globalScope?.document.visibilityState === 'visible') {
+        this.switchToFull()
+      }
+    }, this.featureName, this.ee)
+
     const { error_sampling_rate, sampling_rate, autoStart, block_selector, mask_text_selector, mask_all_inputs, inline_stylesheet, inline_images, collect_fonts } = getConfigurationValue(this.agentIdentifier, 'session_replay')
 
-    this.waitForFlags(['sr']).then(([flagOn]) => {
-      this.entitled = flagOn
+    this.waitForFlags(['srs', 'sr']).then(([srMode, entitled]) => {
+      // waitForFlags now returns the raw value (0, 1) of the flag.
+      // Eventually BCS will report this as (0, 1, 2) for the appropriate mode, and report entitlements separately.
+      // This will need to be fixed to honor that when that change is made. Returning the raw flag value enables that work.
+      this.entitled = !!entitled
       if (!this.entitled && this.recorder?.recording) {
-        this.abort(ABORT_REASONS.ENTITLEMENTS)
         handle(SUPPORTABILITY_METRIC_CHANNEL, ['SessionReplay/EnabledNotEntitled/Detected'], undefined, FEATURE_NAMES.metrics, this.ee)
+        this.recorder?.abort(ABORT_REASONS.ENTITLEMENTS)
         return
       }
-      this.initializeRecording(
-        (Math.random() * 100) < error_sampling_rate,
-        (Math.random() * 100) < sampling_rate
-      )
+      this.initializeRecording(srMode)
     }).then(() => {
-      if (this.mode === MODE.OFF) args?.recorder?.stopRecording() // stop any conservative preload recording launched by instrument
-      sharedChannel.onReplayReady(this.mode) // notify watchers that replay started with the mode
-    })
+      if (this.mode === MODE.OFF) this.recorder?.stopRecording() // stop any conservative preload recording launched by instrument
```

Comment:
These lines are very roundabout. Instead of having it split up, can `initializeRecording` host L124-125 if not entitled and then return?

Reply:
potato potahto, but sure